### PR TITLE
Support Semi-Fungible Amounts with a CopyBag asset-kind

### PR DESCRIPTION
**Pull request by erights**

### CODE REVIEW

**Comment by erights** [comment]
Ready for review.

File: packages/ERTP/src/typeGuards.js [hunk 1]
```diff
@@ -38,7 +38,34 @@ const CopySetValueShape = M.set();
...
+ * from the same asset class.
+ *
+ * If a given value class represents concert tickets, it seems bizarre
+ * that we can form amounts of any key. The hard constraint is that
+ * the code that holds the mint for that asset class---the one associated
+ * with that brand, only mints the items representing the real units
+ * of that asset class as defined by it. Anyone else can put together
+ * an amount expressing, for example, that they "want" some items that
+ * will never be minted. That want will never be satisfied.
+ * "You can't always get..."
```
Comment (dckc) [inline packages/ERTP/src/typeGuards.js:59]: :)

File: packages/store/src/keys/compareKeys.js [hunk 2]
```diff
@@ -117,16 +118,13 @@ export const compareKeys = (left, right) => {
...
-          // Sets of remotables are a particularly bad case. For these, some
-          // kind of hash table (scalar set or map) should be used instead.
-
-          // TODO to get something working, I am currently implementing
-          // only the special case where there are no rank-order ties.
-
           return setCompare(left, right);
         }
+        case 'copyBag': {
+          // TODO write after @gibson042's fix to the copySet text above
```
Comment (dckc) [inline packages/store/src/keys/compareKeys.js:124]: Do we approve PRs with TODOs in them?

**Comment by dckc** [comment]
I'd like to discuss this a bit:

https://github.com/Agoric/agoric-sdk/blob/dc-amount-proptest/packages/ERTP/test/unitTests/test-amountProperties.js

(see also #557)

**Review by dckc — COMMENTED** [review]
a couple warm-up comments / questions...

**Review by dckc — APPROVED** [review]
As we discussed, adding property-based testing from #4366 seems worthwhile.

But this looks good.

**Comment by erights** [comment]
Now that this PR is approved (thanks!), I squashed. But I have not merged yet because this PR is staged on #4305 , which still needs to be reviewed.

**Comment by JimLarson** [comment]
Consider adding `min()` and `max()` properties, which are extremely useful, particularly if subtraction is a partial function. They'll obey the properties outlined in the comment at https://github.com/Agoric/agoric-sdk/blob/3991-standalone-loc/golang/cosmos/types/coin.go#L38

File: packages/store/src/keys/compareKeys.js (hunk 2 above)
Comment (dckc) [reply packages/store/src/keys/compareKeys.js:124]: per discussion with MarkM, yes, we do

**Comment by erights** [comment]
> Consider adding `min()` and `max()` properties, which are extremely useful, particularly if subtraction is a partial function. They'll obey the properties outlined in the comment at https://github.com/Agoric/agoric-sdk/blob/3991-standalone-loc/golang/cosmos/types/coin.go#L38

I agree this would be desirable. But it doesn't need to be in this PR so I filed #4397

**Pull request merged**